include/tvm/relay/expr.h

```diff
@@ -375,4 +376,15 @@ RELAY_DEFINE_NODE_REF(If, IfNode, Expr);
 
 }  // namespace relay
 }  // namespace tvm
+
+namespace std {
+
+template<>
+struct hash<::tvm::relay::Var> {
+  std::size_t operator()(const ::tvm::relay::Var & lv) const {
```

Comment:
@tqchen is it possible to add a generic hasher for `Node` based on contents using `VisitAttrs`?

Reply:
This seems to be just too expensive and we don't want to do that. Note that you can use customized hash struct, instead of having to define hash on a data

Reply:
see https://github.com/dmlc/tvm/blob/master/include/tvm/relay/expr_functor.h#L165

Reply:
It is true, though it is a bit hard to do it for every node, so use unordered_map<SomeNode, Value, NodeHash, NodeEqual> kind of simplifies things